### PR TITLE
Add pod eviction annotations to relevant Observability pods

### PR DESCRIPTION
**Pull request by vmanley**
Jira: [MGDSTRM-9608](https://issues.redhat.com/browse/MGDSTRM-9608)
Adds `"cluster-autoscaler.kubernetes.io/safe-to-evict": "true"` annotation to relevant pods (alertmanager, prometheus and grafana) created by Operator

### CODE REVIEW

**Review by machi1990 — APPROVED** [review]
thanks @vmanley 

Will the change be rolled automatically?

File: controllers/reconcilers/configuration/alertmanager.go [hunk 1]
```diff
@@ -39,64 +40,73 @@ func (r *Reconciler) reconcileAlertmanager(ctx context.Context, cr *v1.Observabi
...
-					"-client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token",
-					"-cookie-secret-file=/etc/proxy/secrets/session_secret",
-					fmt.Sprintf("-openshift-service-account=%v", sa.Name),
-					"-openshift-ca=/etc/pki/tls/cert.pem",
-					"-openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
-					"-skip-auth-regex=^/metrics",
+		alertmanager.Spec = prometheusv1.AlertmanagerSpec{
+			PodMetadata: &prometheusv1.EmbeddedObjectMetadata{
+				Annotations: map[string]string{
+					"cluster-autoscaler.kubernetes.io/safe-to-evict": "true",
```
Comment (pb82) [inline controllers/reconcilers/configuration/alertmanager.go:46]: @machi1990 can you confirm that this is the correct annotation?

File: controllers/reconcilers/configuration/token_refresher.go [hunk 2]
```diff
@@ -160,6 +160,9 @@ func (r *Reconciler) createDeploymentFor(ctx context.Context, cr *v1.Observabili
 						"app.kubernetes.io/name":      config.Name,
 						"app.kubernetes.io/version":   TokenRefresherImageTag,
 					},
+					Annotations: map[string]string{
+						"cluster-autoscaler.kubernetes.io/safe-to-evict": "true",
```
Comment (pb82) [inline controllers/reconcilers/configuration/token_refresher.go:164]: @vmanley is the token refresher using local storage (e.g. an empty dir volumen mount)?

**Comment by pb82** [comment]
@vmanley looks good, not sure if we need the annotation on the token refresher. Also it would be good to test this in an upgrade scenario.

File: controllers/reconcilers/configuration/token_refresher.go (hunk 2 above)
Comment (vmanley) [reply controllers/reconcilers/configuration/token_refresher.go:164]: No storage used. I'll remove annotations. Thanks

**Comment by machi1990** [comment]
> @vmanley looks good, not sure if we need the annotation on the token refresher. Also it would be good to test this in an upgrade scenario.

@vmanley  If you want to create an autoscaled cluster, you can follow the instructions in https://source.redhat.com/groups/public/service-delivery/service_delivery_blog/configuring_cluster_and_machine_pool_autoscaling_in_ocm

**Comment by vmanley** [comment]
> Will the change be rolled automatically?
@machi1990  Annotations would apply after a resync (resync period is one hour)

**Comment by machi1990** [comment]
> > Will the change be rolled automatically?
> > @machi1990  Annotations would apply after a resync (resync period is one hour)

This is great. Thanks

**Comment by vmanley** [comment]
Added annotation to Prometheus, Alertmanager and Grafana pods. No annotation added to operator or catalogSource pods

**Review by pb82 — APPROVED** [review]

**Comment by pb82** [comment]
@vmanley looks good, let's get it merged

**Pull request merged**